File: frontend-angular/app/models/Product.ts
import { Category } from './Category';
export class Product{
   id?: number;
  name?: string;
  description?: string;
  price?: number;
  image?: string;
  imageBytes?: any;
  categoryId?: number;
  categoryDTO?: Category;
  imageUrl?: string | null; // ✅ Add this
}

File: PRODUCT-SERVICE/src/main/java/com/product/service/entities/Product.java
package com.product.service.entities;

import com.product.service.dto.CategoryDTO;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String name;
    private String description;
    private int price;
    private int categoryId;
    private String image;

    @Transient
    private byte[] imageBytes;

    @Transient
   private CategoryDTO categoryDTO;
}
